pages/test-pickup-times/index.tsx

```diff
@@ -0,0 +1,416 @@
+
+  const updateRoute = () => {
+    const queryString = stringify(currentParams)
+    router.push(`/test-pickup-times?${queryString}`)
+  }
+
+  useEffect(updateRoute, [currentParams])
+
+  if (currentParams && scenario !== currentParams.scenario) {
+    const scenarioParams = scenarios[scenario]
+    if (scenarioParams) {
+      setHoldingLocation(scenarioParams.holdingLocation || "")
+      setFulfillment(scenarioParams.fulfillment || "")
+      setDeliveryLocation(scenarioParams.deliveryLocation || "")
+      setRequestTime(scenarioParams.requestTime || "")
+      setCurrentTime(scenarioParams.currentTime || "")
+    }
+  }
+
+  if (
+    currentParams.fulfillment !== fulfillment ||
+    currentParams.holdingLocation !== holdingLocation ||
+    currentParams.deliveryLocation !== deliveryLocation ||
+    currentParams.requestTime !== requestTime ||
+    currentParams.currentTime !== currentTime
+  ) {
+    if (scenario && scenarios[scenario]) {
+      Object.keys(scenarios[scenario]).forEach((key) => {
+        if (scenarios[scenario][key] !== eval(key)) {
+          console.log(
+            "Detected divergence from scenario: ",
+            key,
+            scenarios[scenario][key],
+            eval(key)
+          )
+        }
+      })
+    }
+    setCurrentParams({
+      fulfillment,
+      holdingLocation,
+      deliveryLocation,
+      requestTime,
+      currentTime,
+      scenario,
+    })
+  }
+
+  return (
+    <>
+      <Head>
+        <title>Test Pickup Times | {SITE_NAME}</title>
+      </Head>
+      <Layout>
+        <Heading level="h2" mb="xl" size="heading4">
+          Test Pickup Times
+        </Heading>
+        <Form
+          id="advancedSearchForm"
+          // We are using a post request on advanced search when JS is disabled so that we can build the query
+          // string correctly on the server and redirect the user to the search results.
+          method="post"
+          action={`${BASE_URL}/test-pickup-times`}
+        >
+          <FormRow gap="grid.m">
+            <FormField id="advancedSearchLeft" gap="grid.s">
+              <Select
+                id="scenario"
+                name="scenario"
+                labelText="Preloaded Scenarios"
+                value={scenario}
+                onChange={(e) =>
+                  setScenario((e.target as HTMLInputElement).value)
+                }
+              >
+                {scenarioOptions.map((opt) => {
+                  return (
+                    <option value={opt.value} key={opt.value}>
+                      {opt.label}
+                    </option>
+                  )
+                })}
+              </Select>
+              <HorizontalRule __css={{ margin: 0 }} />
+
+              <TextInput
+                id="holdingLocation"
+                labelText="Item Location"
+                type="text"
+                name="holdingLocation"
+                value={holdingLocation}
+                key="holdingLocation"
```

Comment:
Are these key props being used? They shouldn't be necessary if the elements aren't being rendered inside a loop.

Reply:
Thanks. Fixed